### PR TITLE
Added logic for discipline deleting, changed disciplines list, added generate-service sh script

### PR DESCRIPTION
**Pull request by Yauheni-Verameichyk**

### CODE REVIEW

**Review by Fevor7 — COMMENTED** [review]
I don't have remarks.

**Review by Fevor7 — APPROVED** [review]

File: backend/interview_portal/src/main/java/by/interview/portal/dto/DisciplineDTO.java [hunk 1]
```diff
@@ -19,7 +17,5 @@
 
     private Long parentId;
 
-    private String parentName;
-
-    private Set<UserBaseInfoDTO> disciplineHeadsList;
+    private boolean hasSubItems;
```
Comment (Fevor7) [inline backend/interview_portal/src/main/java/by/interview/portal/dto/DisciplineDTO.java:20]: I recommend using wrapper-objects(boolean - Boolean)

File: backend/interview_portal/src/main/java/by/interview/portal/facade/DisciplineFacade.java [hunk 2]
```diff
@@ -2,16 +2,18 @@
...
-    List<Discipline> findByParentId(Long id);
+    List<DisciplineDTO> findByParentId(Long id);
 
-    void save(DisciplineDTO discipline);
+    void save(DisciplineWithHeadsDTO discipline);
 
-    List<Discipline> findDisciplinesByUser(String login);
+    List<DisciplineDTO> findDisciplinesByUser(String login);
+
+    public void deleteDiscipline(Long id);
```
Comment (Fevor7) [inline backend/interview_portal/src/main/java/by/interview/portal/facade/DisciplineFacade.java:18]: Interface's methods is  public(default)

**Review by Fevor7 — APPROVED** [review]

File: backend/interview_portal/src/main/java/by/interview/portal/facade/DisciplineFacade.java (hunk 2 above)
Comment (Yauheni-Verameichyk) [reply backend/interview_portal/src/main/java/by/interview/portal/facade/DisciplineFacade.java:18]: Agree, I'll fix this later

File: backend/interview_portal/src/main/java/by/interview/portal/dto/DisciplineDTO.java (hunk 1 above)
Comment (Yauheni-Verameichyk) [reply backend/interview_portal/src/main/java/by/interview/portal/dto/DisciplineDTO.java:20]: I disagree here. Why do you recommend this? Any reasons?

**Pull request merged**